Source/WebCore/dom/Range.cpp

```diff
@@ -310,7 +310,8 @@ ExceptionOr<RefPtr<DocumentFragment>> Range::processContents(ActionType action)
         return fragment;
 
     RefPtr<Node> commonRoot = commonAncestorContainer();
-    ASSERT(commonRoot);
+    if (!commonRoot)
+        return Exception { NotFoundError };
```

Comment:
Hm... this doesn't seem right. Chrome & Firefox wouldn't throw an exception in this case.
Instead, setStartAfter/setEndBefore are not setting start/end to x1/x2.

Reply:
Working on it.